addons/website_slides/models/slide_slide.py

```diff
@@ -190,6 +202,47 @@ def _compute_user_info(self):
         for slide in self:
             slide.update(slide_data[slide.id])
 
+    @api.depends('channel_id.slide_ids.is_category', 'channel_id.slide_ids.sequence')
+    def _compute_slide_ids(self):
+        """Will take all the slides of the channel for which the index is higher than the index of this category
+        and lower than the index of the next category"""
+        for slide in self:
```

Comment:
```
child_slides = self.env['slide.slide']
current_category = self.env['slide.slide']
# loop through ordered self: when hitting a new category, previous slides belong to previous category
for slide in self:
  if slide.is_category:
    if current_category:
      current_category.slide_ids = child_slides
      child_slides = self.env['slide.slide']
      current_category = slide
  if not slide.is_category:
    if slide.category_id != current_category:
      slide.category_id = current_category.id
    child_slides |= slide
# ending: don't forget last slides
if current_category and child_slides:
  current_category.child_slides = child_slides
```

Reply:
child_slides = self.env['slide.slide']
current_category = self.env['slide.slide']
# loop through ordered self: when hitting a new category, previous slides belong to previous category
for slide in self:
  if slide.is_category:
    if current_category:
      child_slides.category_id = current_category
    child_slides = self.env['slide.slide']
    current_category = slide
  if not slide.is_category:
    child_slides |= slide
# ending: don't forget last slides
if current_category and child_slides:
  child_slides.category_id = current_category